tutorials/scripting/gdscript/static_typing.rst

```diff
@@ -396,6 +396,71 @@ if you prefer a more readable and reliable, but more verbose syntax.
 ``UNSAFE_*`` warnings make unsafe operations more noticeable, than unsafe lines.
 Currently, ``UNSAFE_*`` warnings do not cover all cases that unsafe lines cover.
 
+Common unsafe operations and their safe counterparts
+----------------------------------------------------
+
+Warnings: ``UNSAFE_PROPERTY_ACCESS`` and ``UNSAFE_METHOD_ACCESS``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this example, we aim to set a property and call a method on an object
+that has a script attached with ``class_name MyScript`` and that ``extends
+Node2D``. If we have a reference to the object as a ``Node2D`` (for instance,
+as it was passed to us by the physics system), we can first check if the
+property and method exist and then set and call them if they do::
+    
+    if "some_property" in node_2d:
+        node_2d.some_property = 20  # gives UNSAFE_PROPERTY_ACCESS warning
+    
+    if node_2d.has_method("some_function"):
+        node_2d.some_function()  # gives UNSAFE_METHOD_ACCESS warning
+
+However, this code will return ``UNSAFE_PROPERTY_ACCESS`` and
+``UNSAFE_METHOD_ACCESS`` warnings as the property and method are not present
+in the referenced type - in this case a ``Node2D``. To make these operations
+safe, you can first check if the object is of type ``MyScript`` using the
+``is`` keyword and then declare a variable with the type ``MyScript`` on
+which you can set its properties and call its methods::
+    
+    if node_2d is MyScript:
+        var my_script: MyScript = node_2d
+        my_script.some_property = 20
+        my_script.some_function()
+
```

Comment:
```suggestion

Alternatively, you can use the ``as`` operator::

    var my_script := node_2d as MyScript
    if my_script:
        my_script.some_property = 20
        my_script.some_function()
```

Reply:
I'll included this alternative in a new commit, but I do want to note that this produces a INFERRED_DECLARATION warning. It doesn't feel entirely right to include example solutions which produce warnings ;)

Reply:
Fair point, but note that `INFERRED_DECLARATION` is added for those who prefer verbose code. This is a strict mode to the second power, we assume that few people enable the warning.

There is also currently no way to specify a metatype: for example if you preload a script or enum into a constant (use `const` as an analogue of `typedef`/`using` in C++). In this case, you are forced to trigger the `INFERRED_DECLARATION` warning and suppress it with `@warning_ignore`.